seismic/ASDFdatabase/asdf_preprocess.py

```diff
@@ -0,0 +1,195 @@
+            # get response object
+            sinv, stationInvCache = getStationInventory(inv, stationInvCache, s)
+
+            st = sta[tag]
+            dayst = Stream()
+            for tr in st:
+                start_time = tr.stats.starttime
+                offset = (UTCDateTime(year=start_time.year, month=start_time.month,
+                                      day=start_time.day) - start_time)
+                for wtr in tr.slide(3600*24, 3600*24, offset=offset, include_partial_windows=True):
+                    wtr = wtr.copy()
+                    dayst += wtr
+                # end for
+            # end for
+            gc.collect()
+
+            # remove response
+            if(sinv):
+                for tr in dayst:
+                    limit_numpy_fft_cache(max_size_in_mb_per_cache=10)
+                    try:
+                        tr.remove_response(sinv, output=instrument_response_output.upper(),
+                                           water_level=water_level)
+                    except Exception as e:
+                        print (e)
+                    # end try
+                    gc.collect()
+                # end for
+            # end if
+
+            # detrend and taper
+            taper_length = 20.0  # seconds
+            for tr in dayst:
+                if tr.stats.npts < 4 * taper_length * tr.stats.sampling_rate:
+                    dayst.remove(tr)
+                else:
+                    tr.detrend(type="demean")
+                    tr.detrend(type="linear")
+                    tr.taper(max_percentage=None, max_length=1.0)
+                # end if
+            # end for
+            gc.collect()
+
+            # apply low-pass filter and create day traces
+            for tr in dayst:
+                tr.filter('lowpass', freq=resample_rate * 0.5, corners=6, zerophase=True)
+                tr.interpolate(resample_rate, method='weighted_average_slopes')
+            # end for
+            gc.collect()
+
+            # add traces
+            for tr in dayst:
+                try:
+                    ods.add_waveforms(tr, tag='raw_recording')
+                except Exception as e:
+                    print (e)
+                    print (tr)
+                # end try
+            # end for
+            #break
+        # end for
+        gc.collect()
+
+        ods.add_stationxml(ids.waveforms[s].StationXML)
+
+        print ('Closing asdf file..')
+        del ods
+
+        #break
+    # end for
+    del ids
+# end func
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.argument('input-asdf', required=True,
+                type=click.Path(exists=True))
+@click.argument('output-folder', required=True,
+                type=click.Path(exists=True))
+@click.option('--resample-rate', default=10,
+              help="Resample rate in Hz; default is 10 Hz")
```

Comment:
You could also use the `show_default=True` flag to have the `click` CLI documentation display the default value.